Easier way to find all meters in a group

Onyi21 opened the pull request:
# Description

All meters in a group/subgroup are displayed on a single line recursively.

Fixes #591

## Type of change

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [x] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

## Limitations

At this time, there is no work remaining to be done on this issue.

AndrewMei99 committed:
Pulling and pushing updated code

AndrewMei99 committed:
Merge https://github.com/OpenEnergyDashboard/OED into development

AndrewMei99 committed:
npm fix

Onyi21 committed:
created the route

Onyi21 committed:
attempted to create the route

Onyi21 committed:
working on #591

Onyi21 committed:
Revert "working on #591"

This reverts commit eccb08a246fe58f214f8ad8614f736a49fe98966.

Onyi21 committed:
worked on #591

huss committed:
fix issues and add code

The layout of the all meters is not yet done correctly

huss committed:
missed trailing comma added by autocomplete

other lint corrected

Onyi21 committed:
displayed all meters correctly

AndrewMei99 committed:
Modified the text wrap styling for all meters being displayed

All meters are displayed on one line

AndrewMei99 committed:
GroupViewCompent npm fix

Turned the deepMeterNames.map function into an arrow function in GroupViewComponent

AndrewMei99 committed:
GroupViewComponent npm fix

huss commented on package-lock.json:
```diff
@@ -124,9 +124,9 @@
-      "version": "7.16.7",
-      "resolved": "https://registry.npmjs.org/@babel/cli/-/cli-7.16.7.tgz",
-      "integrity": "sha512-0iBF+G2Qml0y3mY5dirolyToLSR88a/KB6F2Gm8J/lOnyL8wbEOHak0DHF8gjc9XZGgTDGv/jYXNiapvsYyHTA==",
+      "version": "7.16.8",
```
The commit message says "npm fix" but several say this. (I think you mean a fix for issues found in the checks in the others but am uncertain here.) I'm unclear on why package-lock.json was changed, esp. given no changes to package.json. Can you please explain why and if this is needed. Note this would mean that your PR changes node modules.

Onyi21 replied:
We didn't change anything in the package.json file. Not sure if it's needed

huss replied:
package-lock.json is showing a fair number of changes. I looked at the current version of the development branch and it seems that these are real changes from the current code. Thus, if you didn't intend to change the packages then this change should be removed so the code can be tested with those versions of the packages. This also has to happen to allow merging of the PR since it can only have desired changes. If you need help then let me know.

huss commented on src/client/app/components/groups/GroupViewComponent.tsx:
```diff
@@ -68,6 +77,15 @@ export default class GroupViewComponent extends React.Component<GroupViewProps,
+						<FormattedMessage id='group.all.meters' />:
+						</p>
+							{this.props.deepMeterNames.map ((item, index) => (
+								<span key={`d_${index}`} style={deepMeters}>{(index ? ', ': '') + item}</span>
```
If there are a lot of child meters then the second line onward start with ", ". It would be good if the comma came after the previous item so in this case it was on the preceding line.

huss commented on src/client/app/components/groups/GroupViewComponent.tsx:
```diff
@@ -46,6 +47,14 @@ export default class GroupViewComponent extends React.Component<GroupViewProps,
 			display: renderEditGroupButton ? 'inline' : 'none',
 			paddingLeft: '5px'
 		};
+		const groupAllMeters: React.CSSProperties = {
```
I don't know for sure if this is where it is happening and this is a small point. Would it be possible to move the "All Meters:" label slightly downward so it is a little farther from the "Edit a group" button? Maybe around the vertical spacing between the "All Meters:" label and the list of meters below that.

huss replied:
This was not done but can be fixed at a future time.

huss reviewed (changes requested):
Thanks to @AndrewMei99 and @Onyi21 for doing this work. Overall, it works well where I made a few comments to look at before it is approved.

AndrewMei99 committed:
Package file upstream

AndrewMei99 committed:
Merge https://github.com/OpenEnergyDashboard/OED into development

AndrewMei99 committed:
Fixed comma wrap and package files

Fetched the original package and package-lock files from upstream and also fixed the comma issue when displaying all meters

huss commented:
Changes look good and addressed the needed items. Testing indicates that it works properly.
